system/jlib/jlog.hpp

```diff
@@ -1342,4 +1348,301 @@ interface IComponentLogFileCreator : extends IInterface
+inline bool fetchLogByClass(StringBuffer & returnbuf, IRemoteLogAccess & logAccess, LogMsgClass logclass, LogAccessTimeRange timeRange, StringArray & cols)
+{
+    return fetchLog(returnbuf, logAccess, getClassLogAccessFilter(logclass), timeRange, cols);
+}
+
+//logAccessPluginConfig expected to contain connectivity and log mapping information
+typedef IRemoteLogAccess * (*newLogAccessPluginMethod_t_)(IPropertyTree & logAccessPluginConfig);
+
+class jlib_decl RemoteLogAccessLoader
+{
+public:
+    RemoteLogAccessLoader() = default;
+    ~RemoteLogAccessLoader();
+
+    template <class RemoteLogAccessPlugin>
+    static RemoteLogAccessPlugin * queryRemoteLogAccessor()
+    {
+        Owned<IPropertyTree> logAccessPluginConfig = getGlobalConfig()->getPropTree("logaccess");
+        if (!logAccessPluginConfig)
+        {
+            const char *simulatedGlobalYaml = R"!!(global:
+              logaccess:
+                name: localES
+                type: elasticstack
+                libname: libelasticstacklogaccess.so
+                connection:
+                  protocol: http
+                  host: localhost
+                logmap:
+                  global:
+                    indexsearchpattern: filebeat-7.9.3-*
+                    searchcolumn: message
+                    timestampcolumn: created_ts
+                  workunits:
+                    indexsearchpattern: filebeat-7.9.3-*
+                    searchcolumn: hpcc.log.jobid
+                  components:
+                    searchcolumn: kubernetes.container.name
+                  audience:
+                    searchcolumn: hpcc.log.audience
+                  class:
+                    searchcolumn: hpcc.log.class
+            )!!";
+            Owned<IPropertyTree> testTree = createPTreeFromYAMLString(simulatedGlobalYaml, ipt_none, ptr_ignoreWhiteSpace, nullptr);
+            logAccessPluginConfig.set(testTree->getPropTree("global/logaccess"));
+        }
+
+        const char * methodName = "loadLogAccessPlugin";
+        const char * instFactoryName = "createInstance";
+
+#ifdef _DEBUG
+    StringBuffer thexml;
+    toXML(logAccessPluginConfig, thexml,0,0);
+    DBGLOG("*****%s", thexml.str());
+#endif
+
+        StringBuffer libName;
+        logAccessPluginConfig->getProp("@libname", libName);
+        if (libName.isEmpty())
+        {
+            //If no libname provided default to lib<type>logaccess.so
+            StringBuffer type;
+            logAccessPluginConfig->getProp("@type", type);
+            if (type.isEmpty())
+                throw MakeStringException(-1, "%s library name not specified.", methodName);
+            libName.append("lib").append(type.str()).append("logaccess");
+        }
+
+        //Load the DLL/SO
+        HINSTANCE logAccessPluginLib = LoadSharedObject(libName.str(), true, false);
+        if(logAccessPluginLib == nullptr)
+            throw MakeStringException(-1, "%s cannot load library '%s'", methodName, libName.str());
```

Comment:
trivial: MakeStringException is really deprecated, should use makeStringException or makeStringExceptionV as applicable

Reply:
Taken note, and applying elsewhere as well.